gateway/src/main/scala/filodb/gateway/conversion/InputRecord.scala

```diff
@@ -207,10 +207,62 @@ case class PrometheusInputRecord(tags: Map[String, String],
   val nonMetricShardValues: Seq[String] = nonMetricShardCols.flatMap(tags.get)
   final def getMetric: String = metric
 
-  final def addToBuilder(builder: RecordBuilder): Unit =
+  def addToBuilder(builder: RecordBuilder): Unit =
```

Comment:
I think there is potential to refactor this method to take in the schema as argument. This will remove the redundant classes for each schema type.

Reply:
@vishramachandran @sandeep6189 @alextheimer I will work on this in a later PR. Thanks for reviewing.